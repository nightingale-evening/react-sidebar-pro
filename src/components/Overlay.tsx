import React from 'react';
import styled from 'styled-components';

interface Props {
  onOverlayClick: () => void;
}

const StyledOverlay = styled.div`
  position: fixed;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  background-color: rgb(0, 0, 0, 0.3);
  z-index: 100;
`;

export const Overlay: React.FC<Props> = ({ onOverlayClick }) => {
  return (
    <StyledOverlay
      data-testid="overlay-test-id"
      className="overlay"
      onClick={onOverlayClick}
      onKeyPress={onOverlayClick}
      role="button"
      tabIndex={0}
      aria-label="overlay"
    />
  );
};
