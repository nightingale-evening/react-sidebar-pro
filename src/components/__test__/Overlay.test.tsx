import React from 'react';
import { customRender, screen, fireEvent } from '../../utils/testUtils';
import { Overlay } from '../Overlay';

describe('Overlay', () => {
  it('basic snapshot ', () => {
    const handleOverlayClick = jest.fn();
    const { container } = customRender(
      <Overlay onOverlayClick={handleOverlayClick}>Overlay</Overlay>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should initialize Overlay correctly', async () => {
    const handleOverlayClick = jest.fn();
    customRender(<Overlay onOverlayClick={handleOverlayClick} />);
    const overlayElem = screen.getByTestId('overlay-test-id');

    expect(overlayElem).toBeInTheDocument();
    expect(overlayElem).toHaveClass('overlay');
    expect(overlayElem).toHaveStyle({
      position: 'fixed',
      top: '0px',
      right: '0px',
      bottom: '0px',
      left: '0px',
      'background-color': 'rgb(0, 0, 0, 0.3)',
      'z-index': '100',
    });
  });

  it('should call onOverlayClick when clicked', async () => {
    const handleOverlayClick = jest.fn();
    customRender(<Overlay onOverlayClick={handleOverlayClick} />);
    const overlayElem = screen.getByTestId('overlay-test-id');
    fireEvent.click(overlayElem);

    expect(handleOverlayClick).toHaveBeenCalledTimes(1);
  });
});
