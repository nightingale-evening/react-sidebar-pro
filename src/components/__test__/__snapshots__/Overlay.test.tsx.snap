// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay basic snapshot  1`] = `
<div
  aria-label="overlay"
  class="sc-bdvvtL ljhVXp overlay"
  data-testid="overlay-test-id"
  role="button"
  tabindex="0"
/>
`;
